### PR TITLE
Refactor closet and RCS code. Can't put items in closets on modes other than help

### PR DESCRIPTION
**Pull request by farie82**
## What Does This PR Do
- Makes the code less shit for starters. Dear god was it shit
- Accidently putting away items in lockers while you try to hit somebody in them is quite annoying. Especially if it happens on any mode. Even harm mode.
- Fixes a bug where putting an emag in an open locker/chest would emag it.
- Fixes a bug involving antidrop, a crate and an evil person trying to rig said crate

I've tested using packingpaper. The RCS and an emag on all lockers and chests. Body bags can also be RCSed. Fun stuff which I am gonna leave in.

## Why It's Good For The Game
Less shitcode/bugs the better.
The small feature is mostly a QOL addition. It was the thing I wanted to add but dear god the code was gross.

## Images of changes
emag
Gif featuring the bug. Credit and 20% of the upvotes going to @denghis1

## Changelog
:cl:
tweak: Clicking on an open locker/chest on anything but help mode now won't drop your item in there.
fix: Putting an emag in an open locker/chest won't emag it now
fix: Rigging a crate with a nodrop electropack now won't break it
/:cl:

### CODE REVIEW

File: code/game/objects/structures/crates_lockers/closets.dm [hunk 1]
```diff
@@ -143,98 +145,42 @@
...
-			MouseDrop_T(W:affecting, user)      //act like they were dragged onto the closet
-		if(istype(W,/obj/item/tk_grab))
+			var/obj/item/grab/G = W
+			if(large)
+				MouseDrop_T(G.affecting, user)      //act like they were dragged onto the closet
+			else
+				to_chat(user, "<span class='notice'>The locker is too small to stuff [G.affecting] into!</span>")
+		if(istype(W, /obj/item/tk_grab))
 			return FALSE
+		if(user.a_intent != INTENT_HELP) // Stops you from putting your baton in the closet on accident
```
Comment (farie82) [inline code/game/objects/structures/crates_lockers/closets.dm:161]: The actual feature here

File: code/game/objects/structures/crates_lockers/crates.dm [hunk 2]
```diff
@@ -79,107 +79,31 @@
 	return TRUE
 
 /obj/structure/closet/crate/attackby(obj/item/W, mob/user, params)
-	if(istype(W, /obj/item/rcs) && !src.opened)
-		var/obj/item/rcs/E = W
-		if(E.rcell && (E.rcell.charge >= E.chargecost))
-			if(!is_level_reachable(src.z)) // This is inconsistent with the closet sending code
-				to_chat(user, "<span class='warning'>The rapid-crate-sender can't locate any telepads!</span>")
+	if(!opened)
+		if(istype(W, /obj/item/stack/cable_coil))
```
Comment (SteelSlayer) [inline code/game/objects/structures/crates_lockers/crates.dm:83]: Early returns for these two statements would be nice.

File: code/modules/telesci/rcs.dm [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+	if(!rcell || (rcell.charge < chargecost))
+		to_chat(user, "<span class='warning'>Out of charges.</span>")
+		return
+
+	if(!is_level_reachable(C.z))
+		to_chat(user, "<span class='warning'>The rapid-crate-sender can't locate any telepads!</span>")
+		return
+
+	if(mode == RCS_MODE_CALIBRATED)
+		if(!teleporting)
```
Comment (SteelSlayer) [inline code/modules/telesci/rcs.dm:72]: Early return

File: code/modules/telesci/rcs.dm [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+			var/list/L = list()
+			var/list/areaindex = list()
+			for(var/obj/machinery/telepad_cargo/R in world)
+				if(R.stage == 0)
```
Comment (SteelSlayer) [inline code/modules/telesci/rcs.dm:76]: Early continue might be nice here, something like:
```
if(R.stage)
   continue
```

**Pull request merged**